tests/encodeinput.cpp

```diff
@@ -176,29 +177,54 @@ EncodeOutput* EncodeOutput::create(const char* outputFileName, int width , int h
     if(ext==NULL)
         return NULL;
     ext++;//h264;264;jsv;avc;26l;jvt;ivf
-    if(strcasecmp(ext,"h264")==0 ||
-        strcasecmp(ext,"264")==0 ||
-        strcasecmp(ext,"jsv")==0 ||
-        strcasecmp(ext,"avc")==0 ||
-        strcasecmp(ext,"26l")==0 ||
-        strcasecmp(ext,"jvt")==0 ) {
+
+    if (!strcasecmp(codecName, "AVC")) {
```

Comment:
We'd better make codecName as `optional `not `must`. If you do not set this we will guess from extension.If you set this we will ignore the extension. This code will break simple command line like this
yamencode -i xxx.264 -o xxx.265